script.js

```
@@ -1,28 +1,98 @@
-//You can edit ALL of the code here
+
+let url = `https://api.tvmaze.com/shows/1632/episodes`;
+let selectedShow = 0;
 let episodeLength;
-function setup() {
-  createSelectEpisode();
-  createSearchSection();
-  const allEpisodes = getAllEpisodes();
-  episodeLength = allEpisodes.length;
-  makePageForEpisodes(allEpisodes);
-  searchElement = document.getElementById("searchKey");
-  searchElement.addEventListener("input", search);
-  let SelectElement = document.getElementById("selectEpisode");
-}
-function createEpisodeCode(number, season) {
-  number = number < 10 ? `0${number}` : number;
-  season = season < 10 ? `0${season}` : season;
-  return `S${number}E${season}`;
+let allEpisodes = [];
+
+function setup(url, selectedShow) {
+  makeSectionForSelects();
+  fillSelectByShows(selectedShow);
+  fetch(url)
+    .then((Response) => Response.json())
+    .then(function (data) {
+      allEpisodes = data;
+      episodeLength = allEpisodes.length;
+      makeSearchSection();
+      fillSelectByEpisodes(allEpisodes);
+      makeSectionForEpisodes(allEpisodes);
+    })
+    .catch((error) => console.error(error));
```

@DibaKarimi you can throw an error if you want